### ISSUE TITLE
Docker tag configuration udp service doesn't honor loadbalancer.server.port

### ISSUE THREAD

**Issue by bloeper** [traefik/traefik#8673]
### Welcome!

- [X] Yes, I've searched similar issues on [GitHub](https://github.com/traefik/traefik/issues) and didn't find any.
- [X] Yes, I've searched similar issues on the [Traefik community forum](https://community.containo.us) and didn't find any.

### What did you do?

Via nomad I've deployed a traefik instance and as a test job besides it a pihole service.  
Both are running in docker, managed by nomad/consul. The traefik versions tried are 2.4, 2.5 and 2.6 just as my own sanity check.

Pihole exposes multiple udp ports:
 - 53: dns 
 - 67:dhcp
 
Would like to allow both to be balanced via traefik, but cannot manage it.
It only seems to listen on the "service" port I've defined in my nomad job (in this specific testcase, the dynamic port for dns).

Expected to have 2 udp listeners, 1 tcp and 1 http listener with all their unique ports.
Got 1 tcp and 1 http listener with both unique ports and 2 udp listeners both listening on the same port.

I've tested it with static values in the `port=` statements as well, but it keeps taking the dynamic ports for the dns port (I assume because I've defined that as the "main" service port.  
As stated previously, the tcp/http variants seem to alter the port to whatever I put in them.

It might be I'm doing something wrong, but couldn't figure it out from the docs and since the tcp variant does alter when I switched the port, I suspect it might be a bug.

If I need to give more information please let me know.

Thanks in advance!

### What did you see instead?

The http/tcp tags work as expected, the udp variant only listens on the "service" port which in my testcase was the dns variant.

### What version of Traefik are you using?

Version:      2.5.6
Codename:     brie
Go version:   go1.17.5
Built:        2021-12-22T16:30:52Z
OS/Arch:      linux/amd64

### What is your environment & configuration?

traefik nomad job
``` hcl
job "traefik" {
  datacenters = ["vbox-eu1"]
  type        = "service"
  group "traefik" {
    count = 3
    network {
      port "http" {
        static = 8080
      }
      port "api" {
        static = 8081
      }
    }
    service {
      name = "traefik"

      check {
        name     = "alive"
        type     = "tcp"
        port     = "http"
        interval = "10s"
        timeout  = "2s"
      }
    }
    task "traefik" {
      driver = "docker"

      config {
        image        = "traefik:v2.5"
        network_mode = "host"
        volumes = [
          "local/traefik.toml:/etc/traefik/traefik.toml",
        ]
      }

      template {
        data = <<EOF
[entryPoints]
  [entryPoints.http]
  address = ":8080"
  [entryPoints.traefik]
  address = ":8081"
  [entryPoints.dns-tcp]
  address = ":53"
  [entryPoints.dns-udp]
  address = ":53/udp"
  [entryPoints.dhcp-udp]
  address = ":67/udp"

[api]
  dashboard = true
  insecure  = true

# Enable Consul Catalog configuration backend.
[providers.consulCatalog]
  prefix           = "traefik"
  exposedByDefault = false
  [providers.consulCatalog.endpoint]
    address = "127.0.0.1:8500"
    scheme  = "http"
EOF
        destination = "local/traefik.toml"
      }

      resources {
        cpu    = 100
        memory = 128
      }
    }
  }
}
```
The pihole job:
``` hcl
job "pihole-test" { 
  datacenters = ["vbox-eu1"]
  group "pihole" { 
    count = 2
    task "pihole-server" { 
      service {
        tags = [
          "traefik.enable=true",
          "traefik.http.services.pihole-http.loadbalancer.server.port=${NOMAD_HOST_PORT_http}",

          "traefik.http.routers.pihole-http.service=pihole-http",
          "traefik.http.routers.pihole-http.entrypoints=http",
          "traefik.http.routers.pihole-http.rule=Host(`pihole-test`)",

          "traefik.tcp.routers.pihole-dns-tcp.service=pihole-dns-tcp",
          "traefik.tcp.routers.pihole-dns-tcp.entrypoints=dns-tcp",
          "traefik.tcp.routers.pihole-dns-tcp.rule=HostSNI(`*`)",
          "traefik.tcp.services.pihole-dns-tcp.loadbalancer.server.port=${NOMAD_HOST_PORT_dns_pihole}",

          "traefik.udp.routers.pihole-dns-udp.service=pihole-dns-udp",
          "traefik.udp.routers.pihole-dns-udp.entrypoints=dns-udp",
          "traefik.udp.services.pihole-dns-udp.loadbalancer.server.port=${NOMAD_HOST_PORT_dns_pihole}",

          "traefik.udp.routers.pihole-dhcp-udp.service=pihole-dhcp-udp",
          "traefik.udp.routers.pihole-dhcp-udp.entrypoints=dhcp-udp",
          "traefik.udp.services.pihole-dhcp-udp.loadbalancer.server.port=${NOMAD_HOST_PORT_dhcp_pihole}",
         ]

        port = "dns_pihole"

        check {
          type     = "http"
          path     = "/admin"
          interval = "10s"
          timeout  = "2s"
          port     = "http"
        }
      }
      driver = "docker"       
      config { 
        image = "pihole/pihole:latest" 
        dns_servers = [ 
          "127.0.0.1", 
          "1.1.1.1" 
        ] 
        ports = ["http", "https", "dns_pihole", "dhcp_pihole"]
      } 
      env { 
        TZ = "Amsterdam/Europe" 
      }
    }
    network { 
      port "dns_pihole" {
        to = 53
      } 
      port "dhcp_pihole" {
        to = 67
      } 
      port "http" {
        to = 80
      } 
      port "https" {
        to = 443
      } 
    } 
  }
} 
```

**Comment by tomMoulard** [comment]
Hello @bloeper,

Thanks for your interest in Traefik!

Can you provide the output of your logs (in [DEBUG mode](https://doc.traefik.io/traefik/observability/logs/#level)) in order to view, among other things, the configuration received from nomad ?

**Comment by bloeper** [comment]
Hello @tomMoulard,

Thanks for the quick response and no problem of course.
See the attached logfile of one of the instances.

Kind regards,
- Bloeper
[debug.log](https://github.com/traefik/traefik/files/7808943/debug.log)

**Comment by 3nprob** [comment]
@bloeper This seems to be behaving as expected - the only consul service registered is `pihole-server`, mapped to the TCP port `dns_pihole`.

The idiomatic thing here would be to define one service per port - so you would have 4 services, one mapped to each of your ports, noting that Nomad currently conflates TCP and UDP on the same port so for 53 on both TCP/UDP, that would be one Nomad service for two the two Traefik entrypoints.

**Comment by bloeper** [comment]
Thank you for the response, will check it out with the multiple services.  
If it indeed behaves the way you state, that feature indeed would be important, otherwise we'll need to run the same job multiple times to work around this issue.

**Comment by 3nprob** [comment]
@bloeper regardless, no need to run multiple jobs; there is no 1-to-1 between nomad jobs and consul services.

**Comment by kevinpollet** [comment]
Hello @bloeper,

We were unable to reproduce the issue with Traefik v2.9 because https://github.com/traefik/traefik/pull/9357 fixed it. Therefore, we are closing it, but feel free to comment if you think that we missed something.

Closed by #9357
